### PR TITLE
fix(compiler): make code easier to type check

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: modules/@angular/compiler/src/offline_compiler.ts [hunk 1]
```diff
@@ -129,10 +129,12 @@ export class OfflineCompiler {
 function _resolveViewStatements(compileResult: ViewCompileResult): o.Statement[] {
   compileResult.dependencies.forEach((dep) => {
     if (dep instanceof ViewFactoryDependency) {
-      dep.placeholder.moduleUrl = _ngfactoryModuleUrl(dep.comp);
+      let vfd = <ViewFactoryDependency>dep;
```
Comment (vicb) [inline modules/@angular/compiler/src/offline_compiler.ts]: no need to cast here, use `dep`. TS is smart. 

same thing across the PR.

Comment (vicb) [inline modules/@angular/compiler/src/offline_compiler.ts]: oops that was actually the previous version... why this change ?

Comment (tbosch) [inline modules/@angular/compiler/src/offline_compiler.ts]: To fix G3. We have an old version of Ts in g3 and need to get this change in.

Comment (vicb) [inline modules/@angular/compiler/src/offline_compiler.ts]: Do you know of any plan to update G3 ?
Otherwise we should probably typecheck with the same version in the CI.

Comment (alexeagle) [inline modules/@angular/compiler/src/offline_compiler.ts]: No the g3 version of TS is newer. github/angular/angular is April 9 and g3 is May 12.
See https://github.com/Microsoft/TypeScript/issues/7271
